src/Data/Nat/Base.agda

```diff
@@ -130,6 +132,47 @@ instance
 >-nonZero⁻¹ : ∀ n → .{{NonZero n}} → n > 0
 >-nonZero⁻¹ (suc n) = z<s
 
+-- The property of being a non-zero, non-unit
+
+pattern 2+ n = suc (suc n)
+
+trivial : ℕ → Bool
+trivial 0      = true
+trivial 1      = true
+trivial (2+ _) = false
+
+NonTrivial : Pred ℕ 0ℓ
+NonTrivial = T ∘ not ∘ trivial
+
+instance
+  nonTrivial : ∀ {n} → NonTrivial (2+ n)
+  nonTrivial = _
+
+pattern 1<2+n {n} = s<s (z<s {n})
+
+-- Constructors
+
+n>1⇒nonTrivial : ∀ {n} → 1 < n → NonTrivial n
+n>1⇒nonTrivial 1<2+n = _
+
+nonZero⇒≢1⇒nonTrivial : ∀ {n} → .⦃ NonZero n ⦄ → n ≢ 1 → NonTrivial n
+nonZero⇒≢1⇒nonTrivial {1}    = contradiction refl
+nonZero⇒≢1⇒nonTrivial {2+ _} = _
+
+recompute-nonTrivial : ∀ {n} → .⦃ NonTrivial n ⦄ → NonTrivial n
+recompute-nonTrivial {2+ _} = _
+
+-- Destructors
+
+nonTrivial⇒nonZero : ∀ {n} → .⦃ NonTrivial n ⦄ → NonZero n
```

Comment:
If we *don't* make this an instance (see discussion above), then the `n` should definitely be explicit. I'd prefer it to be an instance though.

Reply:
Yes, I'll see what I can do about hunting this one down... but I do remember it being a pain...

Reply:
`Data.Nat.DivMod` L309:
```agda
/home/james/RESEARCH/Agda-stdlib-dev-v2.0-rc1/src/Data/Nat/DivMod.agda:309,22-29
No instance of type NonTrivial (n * o) was found in scope.
when checking that (*-comm o n) is a valid argument to a function
of type
{n = n₁ : ℕ} {o = o₁ : ℕ} {m = m₁ : ℕ} .⦃ _ = z : NonZero n₁ ⦄
.⦃ _ = z₁ : NonZero o₁ ⦄ →
n₁ ≡ o₁ → m₁ / n₁ ≡ m₁ / o₁
```
trying to typecheck
```agda
m*n/o*n≡m/o : ∀ m n o .⦃ _ : NonZero o ⦄ ⦃ _ : NonZero (o * n) ⦄ →
              m * n / (o * n) ≡ m / o
m*n/o*n≡m/o m n o ⦃ _ ⦄ ⦃ o*n≢0 ⦄ = begin-equality
  m * n / (o * n) ≡⟨ /-congˡ (*-comm m n) ⟩
  n * m / (o * n) ≡⟨ /-congʳ (*-comm o n) ⟩
  n * m / (n * o) ≡⟨ m*n/m*o≡n/o n m o ⟩
  m / o           ∎
  where instance n*o≢0 = subst NonZero (*-comm o n) o*n≢0
```